automate.yaml

```diff
@@ -573,4 +571,4 @@ Outputs:
       Name: !Sub "${AWS::StackName}-DNSName"
   ServerIP:
     Description: The IP Address of the Automate Server
-    Value: !If [UsePublicIp, !GetAtt AutomateServer.PublicIp, !GetAtt AutomateServer.PrivateIp]
+    Value: !GetAtt AutomateServer.PrivateIp
```

Comment:
hey @jeffj254 what's the rationale on this change?   I'm worried it may break existing users who currently need to access via the PublicIp

Reply:
Right now it doesn't look like there's code to ever configure the Automate EC2 instance with a public IP.  The AssociatePublicIpAddress parameter in main.yaml isn't passed to automate.yaml and there's no NetworkInterface section in the Automate EC2::Instance resource to receive it if it was.  Even if there was a way to configure the Automate EC2 instance with a public IP, the UsePublicIp conditional used by this output should probably use the AssociatePublicIpAddress parameter for it's check, rather than the load balancer scheme, since it more directly correlates to if the Automate EC2 instance will have a public IP or not.

Leaving this code in the template will actually cause all stack deployments to fail when using the `internet-facing` scheme since there will never be a public IP on the EC2 instance for this Output to retrieve.  This is what I ran into since we are using internet-facing ELBs with private EC2s.

Reply:
this is where I really wish CF had an equivalent of Terraform's `coalesce` or `compact` functions to pick the Public or Private IP.  We're kinda stuck here aren't we?